commit e42ce0f45ebf20b4c6f89da605cd62b1cd60a9df
Author: Flavio Cruz <flaviocruz@gmail.com>
Date:   Wed Jan 13 00:48:30 2016 +0100

    Fix O_DIRECTORY lookup on trivial translators
    
    * hurd/lookup-retry.c (__hurd_file_name_lookup_retry): Do not remove
    leading slash when `file_name' is "/".

diff --git a/hurd/lookup-retry.c b/hurd/lookup-retry.c
index ef256ac..aee2ba8 100644
--- a/hurd/lookup-retry.c
+++ b/hurd/lookup-retry.c
@@ -62,8 +62,15 @@ __hurd_file_name_lookup_retry (error_t (*use_init_port)
 
   error_t lookup_op (file_t startdir)
     {
-      while (file_name[0] == '/')
-	file_name++;
+      if (file_name[0] == '/' && file_name[1] != '\0')
+	{
+	  while (file_name[1] == '/')
+	    /* Remove double leading slash.  */
+	    file_name++;
+	  if (file_name[1] != '\0')
+	    /* Remove leading slash when we have more than the slash.  */
+	    file_name++;
+	}
 
       return lookup_error ((*lookup) (startdir, file_name, flags, mode,
 				      &doretry, retryname, result));
